services/intersection-api/api/src/test/java/us/dot/its/jpo/ode/api/controllers/live/StompControllerTest.java

```diff
@@ -0,0 +1,111 @@
+package us.dot.its.jpo.ode.api.controllers.live;
+
+import static org.mockito.Mockito.*;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.LineString;
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.map.MapSharedProperties;
+import us.dot.its.jpo.geojsonconverter.pojos.geojson.map.ProcessedMap;
+import us.dot.its.jpo.geojsonconverter.pojos.spat.ProcessedSpat;
+import us.dot.its.jpo.ode.model.OdeBsmData;
+
+public class StompControllerTest {
+
+    @Mock
+    SimpMessagingTemplate brokerMessagingTemplate;
+
+    StompController controller;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        controller = new StompController(brokerMessagingTemplate);
+    }
+
+    @Test
+    void testBroadcastMessage() {
+        controller.broadcastMessage("/topic/test", "hello");
+        verify(brokerMessagingTemplate).convertAndSend("/topic/test", "hello");
+    }
+
+    @Test
+    void testBuildTopicName() {
+        String topic = controller.buildTopicName(123, "processed-spat");
+        assert topic.equals("/live/123/processed-spat");
+    }
+
+    @Test
+    void testBroadcastSpat() throws JsonProcessingException {
+        ProcessedSpat spat = mock(ProcessedSpat.class);
+        when(spat.getIntersectionId()).thenReturn(42);
+
+        controller.broadcastSpat(spat);
+
+        verify(brokerMessagingTemplate).convertAndSend(startsWith("/live/42/processed-spat"), anyString());
```

Comment:
question: we're calling `broadcastSpat` but checking the topic has "processed-spat" in it which is misleading. is this method supposed to just broadcast the processed-spat messages, and should it be named accordingly?